### PR TITLE
fix: Slow sync fails when failing to join an existing mls conversation due to incompatible key packages (WPB-5155)

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Slow sync fails when failing to join an existing mls conversation due to incompatible key packages

### Solutions

We should not propagate non-recoverable errors when joining MLS conversation during slow sync.
So we are ignoring the failure and we just log it instead

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by alexandreferris — APPROVED** [review]

**Review by saleniuk — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]
🚀

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt [hunk 1]
```diff
@@ -65,7 +65,11 @@ internal class JoinExistingMLSConversationsUseCaseImpl(
                                 )
                                 Either.Right(Unit)
                             } else {
-                                Either.Left(it)
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt:68]: can we filter by specific error that we need to ignore?

**Review by MohamadJaara — CHANGES_REQUESTED** [review]

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt (hunk 1 above)
Comment (ohassine) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt:68]: According to user story, we should not propagate non-recoverable errors from this use case.
https://wearezeta.atlassian.net/browse/WPB-5155

**Review by MohamadJaara — APPROVED** [review]

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt (hunk 1 above)
Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt:68]: @ohassine can you please add a comment that this is a non recoverable error

Comment (MohamadJaara) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/JoinExistingMLSConversationsUseCase.kt:68]: @ohassine but network errors are recoverable so can we at least filter out if the user is offline

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
we need to filter out the error user offline errors

**Review by MohamadJaara — APPROVED** [review]
LGTM

**Pull request merged**